sdk/src/types/block/semantic/mod.rs

```diff
@@ -13,32 +13,35 @@ use primitive_types::U256;
 pub use self::{error::TransactionFailureReason, state_transition::StateTransitionVerifier};
 use crate::types::block::{
     address::Address,
-    context_input::{BlockIssuanceCreditContextInput, CommitmentContextInput, RewardContextInput},
-    output::{AccountId, AnchorOutput, ChainId, FoundryId, Output, OutputId, TokenId},
-    payload::signed_transaction::{Transaction, TransactionCapabilityFlag, TransactionSigningHash},
+    context_input::RewardContextInput,
+    output::{feature::Features, AccountId, AnchorOutput, ChainId, FoundryId, Output, OutputId, TokenId},
+    payload::signed_transaction::{Transaction, TransactionCapabilityFlag, TransactionId, TransactionSigningHash},
     protocol::ProtocolParameters,
+    slot::SlotCommitmentId,
     unlock::Unlock,
     Error,
 };
 
 ///
 pub struct SemanticValidationContext<'a> {
     pub(crate) transaction: &'a Transaction,
+    pub(crate) transaction_id: TransactionId,
```

Comment:
Can't you just get this from the transaction?

Reply:
Yes we can, but I'm storing it to avoid having to compute it in a loop

Reply:
But you can do it before the loop in the validate method?